rs/bitcoin/ckbtc/minter/src/lib.rs

```diff
@@ -1126,16 +1127,19 @@ pub fn build_unsigned_transaction_from_inputs(
     // so we simply use 546 satoshi as the minimum amount per output.
     const MIN_OUTPUT_AMOUNT: u64 = 546;
 
-    for (output, fee_share) in unsigned_tx.outputs.iter_mut().zip(fee_shares.iter()) {
-        if output.address != main_address {
-            if output.value <= *fee_share + MIN_OUTPUT_AMOUNT {
-                return Err(BuildTxError::DustOutput {
-                    address: output.address.clone(),
-                    amount: output.value,
-                });
-            }
-            output.value = output.value.saturating_sub(*fee_share);
+    for (output, fee_share) in unsigned_tx
+        .outputs
+        .iter_mut()
+        .zip(fee_shares.iter())
+        .take(num_outputs - 1)
```

Comment:
Instead of matching against main_address, we can just skip the last one, which is more "correct". This comes in handy later when dealing with self-transfers.